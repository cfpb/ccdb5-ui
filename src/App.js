import './css/App.scss';
import { ReactElement } from 'react';
import { Route, BrowserRouter as Router, Routes } from 'react-router-dom';
import { ComplaintDetail } from './components/ComplaintDetail/ComplaintDetail';
import { SearchComponents } from './components/Search/SearchComponents';

/**
 *
 * @returns {ReactElement} Main application component
 */
const App = () => {
  return (
    <Router>
      <Routes>
        {/*
              we need these duplicate routes to match relative path
              /data-research/consumer-complaints/search
              from CF.gov
              local
              which is just the root at localhost:3000/
          */}
        <Route index element={<SearchComponents />} />
        <Route
          path="/data-research/consumer-complaints/search"
          element={<SearchComponents />}
        />
        <Route
          path="/data-research/consumer-complaints/search/detail/:id"
          element={<ComplaintDetail />}
        />
        <Route path="/detail/:id" element={<ComplaintDetail />} />
      </Routes>
    </Router>
  );
};

export default App;
