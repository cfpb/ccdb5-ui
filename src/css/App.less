@import (less) "~cf-core/src/cf-core.less";
@import (less) "~cf-icons/src/cf-icons.less";
@import (less) "~cf-buttons/src/cf-buttons.less";
@import (less) "~cf-forms/src/cf-forms.less";
@import (less) "~cf-grid/src/cf-grid.less";
@import (less) "~cf-layout/src/cf-layout.less";
@import (less) "~cf-typography/src/cf-typography.less";
@import (less) "~cf-pagination/src/cf-pagination.less";
@import (less) "~cf-expandables/src/cf-expandables.less";
@import (less) "~cf-tables/src/cf-tables.less";

@import (less) "./base.less";
@import (less) "../css/notification.less";

// Override for Capital Framework. TODO: remove when migrating to Design System.
// Fix to avoid bleeding into the mega menu.
.o-mega-menu_content-2-list__featured ul {
  margin-bottom: 1.875em;
}

// Fix for erroneously added border below external link in mega menu.
.o-mega-menu_content-link .a-link_text {
  border-bottom-width: 0;
}

// Fix for incorrect header width.
.o-header .wrapper,
.o-footer .wrapper {
  max-width: 1170px !important;
}

// Fix Webpack not packing "approved.svg"

.m-form-field__checkbox .a-checkbox:checked+.a-label:before {
  background-image: url('data:image/svg+xml;charset=utf8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20742.5%201200%22%20width%3D%2216%22%20height%3D%2216%22%20class%3D%22cf-icon-svg%22%3E%3Cpath%20d%3D%22M710%20186c-31.1-18-70.8-7.4-88.8%2023.7v.1L275%20809.5%20114.9%20617.3c-23-27.5-64-31.2-91.6-8.1-27.4%2023-31.2%2063.8-8.3%2091.3l219.8%20263.8c12.3%2014.8%2030.6%2023.4%2049.9%2023.4%201.8%200%203.7-.1%205.6-.2%2021.2-1.8%2040.1-13.9%2050.7-32.3l392.8-680.4c17.9-31.1%207.3-70.9-23.8-88.8z%22%2F%3E%3C%2Fsvg%3E%0A');
}

// App-wide definitions

.layout-row {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.layout-column {
  display: flex;
  flex-direction: column;
}

.flex-fixed {
  flex: none;
}

.flex-all {
  flex: 1 0 auto;
}

.body-copy {
  font-size: 16px;
}

.cols {
  .grid_nested-col-group();
}

.col:before {
  display: block;

  width: 100%;
  content: "";
}

.col-1 {
  .grid_column(1);
}

.col-2 {
  .grid_column(2);
}

.col-3 {
  .grid_column(3);
}

.col-4 {
  .grid_column(4);
}

.col-6 {
  .grid_column(6);
}

.col-8 {
  .grid_column(8);
}

.col-9 {
  .grid_column(9);
}

.col-10 {
  .grid_column(10);
}

.col-12 {
  .grid_column(12);
}

// Overrides

.content__1-3 {
  margin-bottom: @grid_gutter-width * 2;

  .content_wrapper {
    padding-left: 0;
    padding-right: 0;
    border: 1px solid @gray-40;
    border-right: none;
    max-width: initial;
    @media @phone {
      border: none;
    }
  }

  .content_sidebar {
    width: 420px;
    padding: 0;
    border: 0;
    margin: 0;
  }

  .content_main {
    width: ~"calc(100% - 420px)";
    padding: 0;
    border: 0;

    &:after {
      border-left: 0;
    }
  }
}

// gutters
.wrapper__match-content {
  @media @tablet {
    padding-left: 20px;
    padding-right: 20px;
  }
  @media @desktop {
    padding-left: 30px;
    padding-right: 30px;
  }
  @media @superwide {
    padding-left: 80px;
    padding-right: 80px;
  }
}

@media @tablet-lg {
  .content__1-3 {
    .content_sidebar {
      width: 305px;
    }

    .content_main {
      width: ~"calc(100% - 305px)";
    }
  }
}

@media @tablet-sm {
  .content__1-3 {
    .content_main {
      width: 100%;
    }
    .content_sidebar {
        &.map,
        &.list {
          width: 100%;
        }
    }
  }
  .filter-button {
    display: block;
    margin: 20px 10px;
    button {
      width: 100%;
    }
  }
}


@media @phone {
  .filter-button {
    display: block;
    margin: 20px 0px;
    button {
      width: 100%;
    }
  }
  .content__1-3 {
    margin-bottom: 0;
    .content {
      &_sidebar,
      &_main {
        width: 100%;
      }

    }
  }
}

body {
  a,
  .a-btn__link {
    color: @link-text;
    &:hover {
      color: @link-text-hover;
    }
  }
}

::-ms-clear {
  display: none;
}

// Icon color in error message when entering an incorrect date.
.a-error-message .cf-icon-delete-round {
  color: @red;
}
@import (less) "../components/Print/print.less";
