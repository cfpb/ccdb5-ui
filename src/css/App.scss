// Import Design System Components.
@use '@cfpb/cfpb-design-system/src/abstracts' as *;

// no atomic components
@use 'base' as *;
@use 'components/Print/print' as *;


// Icon font path
$cf-icon-path: '../css/fonts';

// Webfont variables
// This is the path for self-hosted fonts.
$cf-fonts-path: '/static/fonts';

// Fix for incorrect header width.
.o-header .wrapper,
.o-footer .wrapper {
  max-width: 1170px !important;
}

// App-wide definitions

.layout-row {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.layout-column {
  display: flex;
  flex-direction: column;
}

.flex-fixed {
  flex: none;
}

.flex-all {
  flex: 1 0 auto;
}

.body-copy {
  font-size: 16px;
}

.cols {
  @include u-grid-nested-col-group;
}

.col::before {
  display: block;

  width: 100%;
  content: '';
}

.col-1 {
  @include u-grid-column(1); 
}

.col-2 {
  @include u-grid-column(2);
}

.col-3 {
  @include u-grid-column(3);
}

.col-4 {
  @include u-grid-column(4);
}

.col-6 {
  @include u-grid-column(6);
}

.col-8 {
  @include u-grid-column(8);
}

.col-9 {
  @include u-grid-column(9);
}

.col-10 {
  @include u-grid-column(10);
}

.col-12 {
  @include u-grid-column(12);
}

// Overrides

.content--1-3.ccdb-content {
  margin-bottom: $grid-gutter-width * 2;

  .content__wrapper {
    padding-left: 0;
    padding-right: 0;
    border: 1px solid var(--gray-40);
    border-right: none;
    max-width: initial;

    @media screen and (max-width: $layout-breakpoint-phone-max) {
      border: none;
    }
  }

  .content__sidebar {
    width: 420px; 
    padding: 0;
    border: 0;
    margin: 0;
  }

  .content__main {
    width: calc(100% - 420px);
    padding: 0;
    border: 0;

    &::after {
      border-left: 0;
    }
  }
}

// gutters
.wrapper-match-content {
  @media screen and (min-width: $layout-breakpoint-tablet-min) and (max-width: $layout-breakpoint-tablet-max) {
    padding-left: 20px;
    padding-right: 20px;
  }

  @media screen and (min-width: $layout-breakpoint-desktop-min) {
    padding-left: 30px;
    padding-right: 30px;
  }

  @media screen and (min-width: $layout-breakpoint-superwide-min) {
    padding-left: 80px;
    padding-right: 80px;
  }
}

@media screen and (min-width: $layout-breakpoint-tablet-lg-min) and (max-width: $layout-breakpoint-tablet-lg-max) {
  .content--1-3.ccdb-content {
    .content__sidebar {
      width: 305px;
    }

    .content__main {
      width: calc(100% - 305px);
    }
  }
}

@media screen and (min-width: $layout-breakpoint-tablet-sm-min) and (max-width: $layout-breakpoint-tablet-sm-max) {
  .content--1-3.ccdb-content {
    .content__main {
      width: 100%;
    }
    .content__sidebar {
      &.map,
      &.list {
        width: 100%;
      }
    }
  }
  .filter-button {
    display: block;
    margin: 20px 10px;
    button {
      width: 100%;
    }
  }
}

@media screen and (max-width: $layout-breakpoint-phone-max) {
  .filter-button {
    display: block;
    margin: 20px 0;
    button {
      width: 100%;
    }
  }
  .content--1-3.ccdb-content {
    margin-bottom: 0;
    .content {
      &__sidebar,
      &__main {
        width: 100%;
      }
    }
  }
}


::-ms-clear {
  display: none;
}