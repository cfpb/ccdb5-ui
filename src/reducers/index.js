import aggReducer  from './aggs/aggs';
import detailReducer from './detail/detail';
import mapReducer from './map/map';
import queryReducer from './query/query';
import resultsReducer from './results/results';
import trendsReducer from './trends/trends';
import viewReducer from './view/view';

import { configureStore } from "@reduxjs/toolkit";


export default configureStore({
  reducer: {
    aggregations: aggReducer,
    detail: detailReducer,
    map: mapReducer,
    query: queryReducer,
    results: resultsReducer,
    trends: trendsReducer,
    view: viewReducer
  }

})
