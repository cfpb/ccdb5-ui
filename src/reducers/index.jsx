import aggs from './aggs'
import { combineReducers } from 'redux'
import detail from './detail'
import map from './map'
import modal from './modal'
import query from './query'
import results from './results'
import trends from './trends'
import view from './view'

export default combineReducers( {
  aggs,
  detail,
  map,
  modal,
  query,
  results,
  trends,
  view
} )
