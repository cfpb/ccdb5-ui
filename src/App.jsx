// Required so that the expose-loader test works which moves the ReactDOM
// variable into the global space
// eslint-disable-next-line
import ReactDOM from 'react-dom';

import React from 'react';
import { Provider } from 'react-redux'
import { IntlProvider } from 'react-intl';
import { createStore, applyMiddleware } from 'redux'
import thunkMiddleware from 'redux-thunk'

import { composeWithDevTools } from 'redux-devtools-extension';

import {
  BrowserRouter as Router,
  Route,
  Switch
} from 'react-router-dom'


import reducers from './reducers'
import FilterPanel from './FilterPanel';
import Hero from './Hero';
import SearchPanel from './SearchPanel';
import ResultsPanel from './ResultsPanel';
import ComplaintDetail from './ComplaintDetail';
import RootModal from './Dialogs/RootModal'
import UrlBarSynch from './UrlBarSynch';
import './App.less';

const middleware = [thunkMiddleware];

const composeEnhancers = composeWithDevTools({
  // required for redux-devtools-extension
  // Specify name here, actionsBlacklist, actionsCreators and other options if needed
});

// required format for redux-devtools-extension
const store = createStore(reducers, /* preloadedState, */ composeEnhancers(
  applyMiddleware(...middleware),
  // other store enhancers if any
));


export class SearchComponents extends React.Component {
  render() {
    return (
      <IntlProvider locale="en">
        <main className="content content__1-3" role="main">
          <UrlBarSynch />
          <Hero />
          <div className="content_wrapper">
            <SearchPanel />
            <aside className="content_sidebar">
              <FilterPanel />
            </aside>
            <ResultsPanel className="content_main" />
          </div>
          <RootModal />
        </main>
      </IntlProvider>
    )
  }
}

export class DetailComponents extends React.Component {
  render() {
    const complaint_id = this.props.match.params.id;

    return (
      <IntlProvider locale="en">
          <ComplaintDetail complaint_id={complaint_id}/>
      </IntlProvider>
    )
  }
}

export class App extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <Router>
          <Switch>
            <Route path="*/detail/:id" component={DetailComponents}/>
            <Route path="/" component={SearchComponents}/>
          </Switch>
        </Router>
      </Provider>
    );
  }
}
