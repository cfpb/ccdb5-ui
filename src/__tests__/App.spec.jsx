import { App, DetailComponents } from '../App';
import configureMockStore from 'redux-mock-store'
import { defaultQuery } from '../reducers/query'
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux'
import React from 'react';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk'

describe('initial state', () => {
  it('renders without crashing', () => {

    // set the date so snapshot will always be the same.
    const DATE_TO_USE = new Date('2016');
    const _Date = Date;
    global.Date = jest.fn(() => DATE_TO_USE);
    global.Date.UTC = _Date.UTC;
    global.Date.parse = _Date.parse;
    global.Date.now = _Date.now;
    defaultQuery.searchText = 'foo';

    const target = renderer.create(
      <MemoryRouter initialEntries={[ '/' ]}>
        <App />
      </MemoryRouter>
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();

  });

  it('renders the detail route', () => {
    const middlewares = [thunk]
    const mockStore = configureMockStore(middlewares)
    const store = mockStore({
      detail: { data: {}, error: '' }
    })

    const match = { params: { id: '1234' } };
    const detailTarget = renderer.create(
      <MemoryRouter initialEntries={[ '/detail/1234' ]}>
        <Provider store={store}>
          <DetailComponents match={ match }/>
        </Provider>
      </MemoryRouter>
    );

    let detailTree = detailTarget.toJSON();
    expect(detailTree).toMatchSnapshot();
  });
});
