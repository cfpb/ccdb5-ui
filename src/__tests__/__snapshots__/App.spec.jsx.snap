exports[`initial state renders the detail route 1`] = `
<main
  role="main">
  <section
    className="card-container">
    <nav
      className="layout-row">
      <div
        className="back-to-search flex-fixed">
        <button
          className="a-btn a-btn__link"
          onClick={[Function]}>
          <span
            className="cf-icon cf-icon-left" />
          Back to search results
        </button>
      </div>
      <div
        className="meaning flex-fixed">
        <a
          href="https://www.consumerfinance.gov/complaint/data-use/"
          target="_blank">
          What do all these data points mean?
        </a>
      </div>
    </nav>
    <section
      className="light-box">
      <div
        className="loading-box">
        <span
          className="cf-icon cf-icon-update cf-icon__before cf-icon__spin" />
        <span>
          This page is loading
        </span>
      </div>
    </section>
  </section>
</main>
`;

exports[`initial state renders without crashing 1`] = `
<main
  className="content content__1-3"
  role="main">
  <header
    className="content_hero">
    <h1
      className="content-header">
      Consumer Complaint Database
    </h1>
    <h4
      className="content-header">
      Beta
    </h4>
    <p>
      Consumer complaints are added to this public database after the company has responded to the complaint, confirming a commercial relationship with the consumer, or after they\'ve had the complaint for 15 calendar days, whichever comes first. We donâ€™t verify all the facts alleged in complaints, but we do give companies the opportunity to publicly respond to complaints by selecting responses from a pre-populated list. Company-level information should be considered in the context of company size and/or market share.
    </p>
    <ul
      className="m-list m-list__horizontal">
      <li
        className="m-list_item">
        <button
          className="a-btn a-btn__link"
          onClick={[Function]}>
          More about the complaint database
        </button>
      </li>
      <li
        className="m-list_item">
        <a
          href="https://www.consumerfinance.gov/complaint/data-use/"
          target="_blank">
          How we use complaint data
        </a>
      </li>
      <li
        className="m-list_item">
        <a
          href="https://cfpb.github.io/api/ccdb/"
          target="_blank">
          Technical documentation
        </a>
      </li>
      <li
        className="m-list_item">
        <a
          href="https://data.consumerfinance.gov/dataset/Consumer-Complaints/s6ew-h6mp"
          target="_blank">
          View complaint data in Socrata
        </a>
      </li>
    </ul>
  </header>
  <div
    className="content_wrapper">
    <div
      className="search-panel">
      <h2>
        Search complaint data
      </h2>
      <div>
        <nav
          className="search-bar">
          <form
            action=""
            onSubmit={[Function]}>
            <h4>
              Search Within
            </h4>
            <div
              className="layout-row">
              <div
                className="cf-select flex-fixed">
                <select
                  aria-label="Choose which field will be searched"
                  id="searchField"
                  onChange={[Function]}
                  value="all">
                  <optgroup
                    label="Search Within">
                    <option
                      value="all">
                      All Data
                    </option>
                    <option
                      value="company">
                      Company Name
                    </option>
                    <option
                      value="complaint_what_happened">
                      Narratives
                    </option>
                  </optgroup>
                </select>
              </div>
              <div
                className="flex-all typeahead-portal">
                <section
                  className="typeahead "
                  onBlur={[Function]}
                  onFocus={[Function]}>
                  <div
                    className="m-btn-inside-input">
                    <input
                      aria-label="The term to search for"
                      autoComplete="off"
                      className="a-text-input"
                      id="searchText"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Enter your search term(s)"
                      type="text"
                      value="foo" />
                    <button
                      className="a-btn a-btn__link"
                      onClick={[Function]}>
                      <span
                        className="cf-icon cf-icon-delete" />
                      Clear
                    </button>
                  </div>
                </section>
              </div>
              <button
                className="a-btn flex-fixed"
                type="submit">
                Search
                <span
                  className="a-btn_icon a-btn_icon__on-right cf-icon cf-icon__after cf-icon-search" />
              </button>
              <div
                className="advanced-container flex-fixed">
                <a
                  className="a-btn a-btn__link o-expandable_cue-open"
                  onClick={[Function]}>
                  Show advanced search tips
                </a>
              </div>
            </div>
          </form>
        </nav>
      </div>
    </div>
    <aside
      className="content_sidebar">
      <section
        className="filter-panel">
        <h3>
          Filter results by...
        </h3>
        <section
          className="single-checkbox">
          <h4>
            Only show complaints with narratives?
          </h4>
          <div
            className="m-form-field m-form-field__checkbox">
            <input
              checked={false}
              className="a-checkbox"
              id="theCheckbox"
              onClick={[Function]}
              type="checkbox"
              value="has_narrative" />
            <label
              className="a-label"
              htmlFor="theCheckbox">
              Yes
            </label>
          </div>
        </section>
        <hr />
        <section
          className="o-expandable aggregation date-filter">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              Date CFPB received the complaint
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p />
          <div>
            <ul
              className="date-inputs">
              <li>
                <label
                  className="a-label a-label__heading body-copy">
                  From
                </label>
                <div
                  className="m-btn-inside-input">
                  <input
                    aria-describedby="input-error_message-from"
                    className="a-text-input"
                    max={null}
                    min={2014-01-01T12:00:00.000Z}
                    onChange={[Function]}
                    placeholder="mm/dd/yyyy"
                    size="15"
                    value="" />
                </div>
              </li>
              <li>
                <label
                  className="a-label a-label__heading body-copy">
                  Through
                </label>
                <div
                  className="m-btn-inside-input">
                  <input
                    aria-describedby="input-error_message-through"
                    className="a-text-input"
                    max={null}
                    min={2014-01-01T12:00:00.000Z}
                    onChange={[Function]}
                    placeholder="mm/dd/yyyy"
                    size="15"
                    value="" />
                </div>
              </li>
            </ul>
          </div>
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              Matched company name
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p>
            The company name as it appears in our complaint system, which may be different than the name the consumer provided in their complaint
          </p>
          <section
            className="typeahead "
            onBlur={[Function]}
            onFocus={[Function]}>
            <div
              className="m-btn-inside-input">
              <input
                autoComplete="off"
                className="a-text-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Enter company name"
                type="text"
                value="" />
            </div>
          </section>
          <ul />
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              Product / sub-product
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p>
            The type of product and sub-product the consumer identified in the complaint
          </p>
          <p>
            <a
              className="a-btn__link"
              href="http://files.consumerfinance.gov/f/documents/201704_cfpb_Summary_of_Product_and_Sub-product_Changes.pdf"
              target="_blank">
              Recent changes to products and sub-products
            </a>
          </p>
          <div>
            <ul />
          </div>
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              Issue / sub-issue
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p>
            The type of issue and sub-issue the consumer identified in the complaint
          </p>
          <section
            className="typeahead "
            onBlur={[Function]}
            onFocus={[Function]}>
            <div
              className="m-btn-inside-input">
              <input
                autoComplete="off"
                className="a-text-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Enter name of issue"
                type="text"
                value="" />
            </div>
          </section>
          <div>
            <ul />
          </div>
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              State
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p>
            The state of the mailing address provided by the consumer
          </p>
          <section
            className="typeahead "
            onBlur={[Function]}
            onFocus={[Function]}>
            <div
              className="m-btn-inside-input">
              <input
                autoComplete="off"
                className="a-text-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Enter state name or abbreviation"
                type="text"
                value="" />
            </div>
          </section>
          <ul />
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              Zip code
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p>
            The mailing ZIP code provided by the consumer
          </p>
          <section
            className="typeahead "
            onBlur={[Function]}
            onFocus={[Function]}>
            <div
              className="m-btn-inside-input">
              <input
                autoComplete="off"
                className="a-text-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Enter first three digits of ZIP code"
                type="text"
                value="" />
            </div>
          </section>
          <ul />
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              Did company provide a timely response?
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
                onClick={[Function]}>
                Show
                <span
                  className="cf-icon cf-icon-plus-round" />
              </button>
            </span>
          </div>
          <p />
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              Company response
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
                onClick={[Function]}>
                Show
                <span
                  className="cf-icon cf-icon-plus-round" />
              </button>
            </span>
          </div>
          <p>
            How the company responded to the complaint
          </p>
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              Company public response
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
                onClick={[Function]}>
                Show
                <span
                  className="cf-icon cf-icon-plus-round" />
              </button>
            </span>
          </div>
          <p>
            The company\'s optional public-facing response to a consumer\'s complaint
          </p>
        </section>
        <hr />
        <section
          className="o-expandable aggregation date-filter">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              Date complaint sent to company
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
                onClick={[Function]}>
                Hide
                <span
                  className="cf-icon cf-icon-minus-round" />
              </button>
            </span>
          </div>
          <p />
          <div>
            <ul
              className="date-inputs">
              <li>
                <label
                  className="a-label a-label__heading body-copy">
                  From
                </label>
                <div
                  className="m-btn-inside-input">
                  <input
                    aria-describedby="input-error_message-from"
                    className="a-text-input"
                    max={null}
                    min={2014-01-01T12:00:00.000Z}
                    onChange={[Function]}
                    placeholder="mm/dd/yyyy"
                    size="15"
                    value="" />
                </div>
              </li>
              <li>
                <label
                  className="a-label a-label__heading body-copy">
                  Through
                </label>
                <div
                  className="m-btn-inside-input">
                  <input
                    aria-describedby="input-error_message-through"
                    className="a-text-input"
                    max={null}
                    min={2014-01-01T12:00:00.000Z}
                    onChange={[Function]}
                    placeholder="mm/dd/yyyy"
                    size="15"
                    value="" />
                </div>
              </li>
            </ul>
          </div>
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              Consumer consent
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
                onClick={[Function]}>
                Show
                <span
                  className="cf-icon cf-icon-plus-round" />
              </button>
            </span>
          </div>
          <p>
            Whether a consumer opted to publish their compaint narrative
          </p>
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              How did the consumer submit the complaint to the CFPB?
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
                onClick={[Function]}>
                Show
                <span
                  className="cf-icon cf-icon-plus-round" />
              </button>
            </span>
          </div>
          <p />
        </section>
        <hr />
        <section
          className="o-expandable aggregation">
          <div
            className="o-expandable_header o-expandable_target">
            <h4
              className="o-expandable_header-left o-expandable_label">
              Tags
            </h4>
            <span
              className="o-expandable_header-right o-expandable_link">
              <button
                className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
                onClick={[Function]}>
                Show
                <span
                  className="cf-icon cf-icon-plus-round" />
              </button>
            </span>
          </div>
          <p>
            Data that supports easier searching and sorting of complaints submitted by or on behalf of older Americans and/or servicemembers
          </p>
        </section>
      </section>
    </aside>
    <section
      className="results-panel content_main">
      <summary
        className="action-bar">
        <div>
          <h2>
            ShowingÂ 
            <span>
              0
            </span>
            Â total complaints
          </h2>
        </div>
        <div
          className="layout-row">
          <div
            className="cf-select flex-fixed">
            <select
              id="choose-size"
              onChange={[Function]}
              value={25}>
              <option
                value={10}>
                Show 
                10
                 results
              </option>
              <option
                value={25}>
                Show 
                25
                 results
              </option>
              <option
                value={50}>
                Show 
                50
                 results
              </option>
              <option
                value={100}>
                Show 
                100
                 results
              </option>
            </select>
          </div>
          <div
            className="cf-select flex-fixed">
            <select
              id="choose-sort"
              onChange={[Function]}
              value="created_date_desc">
              <option
                value="created_date_desc">
                Sort by newest to oldest
              </option>
              <option
                value="created_date_asc">
                Sort by oldest to newest
              </option>
              <option
                value="relevance_desc">
                Sort by relevance
              </option>
              <option
                value="relevance_asc">
                Sort by relevance (asc)
              </option>
            </select>
          </div>
          <h4
            className="flex-all">
            <button
              className="a-btn a-btn__link"
              onClick={[Function]}>
              Export results
            </button>
          </h4>
        </div>
      </summary>
      <h2>
        No results were found for your search
      </h2>
      <section
        className="light-box">
        <div
          className="loading-box">
          <span
            className="cf-icon cf-icon-update cf-icon__before cf-icon__spin" />
          <span>
            This page is loading
          </span>
        </div>
      </section>
    </section>
  </div>
  <span />
</main>
`;
